test/int/project/project_test.go

```diff
@@ -0,0 +1,100 @@
+package project
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/kube"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/testutil"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("AtlasProject", func() {
+	const timeout = time.Second * 20
+	const interval = time.Second * 1
+
+	var (
+		namespace        corev1.Namespace
+		connectionSecret corev1.Secret
+		projectID        string
+	)
+
+	BeforeEach(func() {
```

Comment:
before: creating namespace and a connection secret.
after: deleting the project if it was created and removing the namespace